### ISSUE TITLE
PI Docs improvements after EthTokyo

### ISSUE THREAD

**Issue by nambrot** [hyperlane-xyz/hyperlane-monorepo#2112]
Problems:
            - For some people, lack of continuity and hard to follow what’s going on and jumping around different pages
            - No definition of setup for yarn and node
            - not sure if I should keep anvil networks in the chain metadata or add to it. also should it maybe show optional fields like etherscan (for nicer error logging)?
            - Not clear to folks how to set overrides like manual gasLimit (often necessary for not-quite-EVM-networks)
            - we don't seem to be indicating to folks confirmations and why it matters and we default to 1 in the absence?
            - Logs mention storage oracles but very hard to find anything on the docs about what the IGP payment set is for the deploy
            - A bit unclear on what the test does (other than giving binary success conditions)
            - Docs and code are very inconsistent, there are talks about anvil1 and anvil2, then just anvil with goerli and sepilia
            - default --key is not consistent with other sections
            - Folks often miss that they have to have the validators and relayers running for messages to be delivered

Possible Solutions (not exhaustive):
            - Collapse “deploy hyperlane” into a single page
            - would probably like some intro/context as to why i want to do this and the benefits once i complete this
            - do like the high level overview of the steps
            - Yarn/node setup section
            - the link for chain metadata should imo point to the file the user should update, not the chain metadata definition in the sdk
            - The comments in chains.ts should be more helpful to allow the deployer to understand what fields they want to set and why (i.e. block explorer, overrides, etc.)
            - Remove anvil1/2 from the chains.ts and instead copy over a file for CI, and have docs and code mostly simulate a deploy to anvil and goerli/sepolia (or maybe just sepolia since goerli etc is hard to get)
            - Document IGP setup for PI deploy
            - Add recaps to each section
            - Explain what test-messages does, who the recipient is, that validators and relayers need to run and we are polling for delivery
            - Tell the deployer about the explorer and how it can find messages to/from their PI chain (by adding metadata)

**Comment by nambrot** [comment]
https://discord.com/channels/935678348330434570/1098862203458834475/1099455112776265728

It seems not obvious that you have to set the ism when you build your own smart contracts and deploy them

**Comment by asaj** [comment]
> It seems not obvious that you have to set the ism when you build your own smart contracts and deploy them

@nambrot I'm not really sure what to do with this.

**Comment by asaj** [comment]
> - A bit unclear on what the test does (other than giving binary success conditions)

From the docs:
```You can check everything is working correctly by sending a test message between each pair of chains.```

What about this is unclear?

**Comment by asaj** [comment]
> - default --key is not consistent with other sections
What does this mean?

**Comment by nambrot** [comment]
> > It seems not obvious that you have to set the ism when you build your own smart contracts and deploy them
> 
> @nambrot I'm not really sure what to do with this.

Did you see the linked Discord thread? In this case, a developer has deployed Hyperlane on a PI chain, was successfully running the test messages script but then unable to deploy their own contracts and test message passing between them. The root cause of their problem was that they did not realize that their contracts have to specify an ISM. Looking at https://docs.hyperlane.xyz/docs/apis-and-sdks/messaging-api/receive, or https://docs.hyperlane.xyz/docs/build-with-hyperlane/quickstarts/messaging or https://docs.hyperlane.xyz/docs/deploy/deploy-hyperlane/send-test-messages, i don't blame the developer for it not being clear that they would have to specify an ISM to accomplish their task.

I would suggest to the general messaging docs to give a hint that if you send from a PI chain to a default chain, they will need to set the ISM and cannot rely on the default ISM.



> > * A bit unclear on what the test does (other than giving binary success conditions)
> 
> From the docs: `You can check everything is working correctly by sending a test message between each pair of chains.`
> 
> What about this is unclear?

For the send-test-messages section of the PI guide, I think it would be helpful to be more verbose in what the test does, i.e. it dispatches a message from the origin chain (probably ideally the PI chain for illustrative purposes) to a TestRecipient on the destination chain that is configured with the ISM that we deployed (and explain why we can't use the default). Then say that the validator observes the mailbox root, provides the signature, the relayer indexes the message, fetches the signature and processes. (this point addresses the fact that some folks didn't realize that the agents have to be running for messages to be processed).

I think many folks will have sufficient background to understand the various pieces as they do the PI guide of what is happening, but i think it's worth educating and solidifying the mental model for the developer at the end with the test.


> > * default --key is not consistent with other sections
> >   What does this mean?

Sorry i should have been more clear, but the key parameter in https://docs.hyperlane.xyz/docs/deploy/deploy-hyperlane/send-test-messages is different from the one in https://docs.hyperlane.xyz/docs/deploy/deploy-hyperlane/deploy-contracts. Maybe we just use an ENV var here?

**Comment by asaj** [comment]
@nambrot I'm having trouble understanding which of these are things that people actually had trouble with vs things that you think would be improvements

For example, with that the private keys in the example commends being different, what friction did we actually observe that was caused by this?

**Comment by nambrot** [comment]
> @nambrot I'm having trouble understanding which of these are things that people actually had trouble with vs things that you think would be improvements
> 
> For example, with that the private keys in the example commends being different, what friction did we actually observe that was caused by this?

There was no problem in that specific example, it's something that was suggested during an internal run https://www.notion.so/Permissionless-Interoperability-Deploy-DX-Game-Yorke-6e0bec56e36647a6a7136bcb3bd1726f?pvs=4#3aa0c7057c234fefb27150afb1348a61

**Comment by asaj** [comment]
Okay, so which of these were actual things that folks ran into that created friction?

**Comment by nambrot** [comment]
Some related feedback in https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2153 that might be worth bundling here

**Comment by nambrot** [comment]
Closing with the only remaining thing to work to be https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2234
